### PR TITLE
[Docathon][Update Doc No.18] Update optimizer_en.rst

### PR DESCRIPTION
**Pull request by cangtianhuang**
Update optimizer_en.rst, add API references for AdamW, Lamb, LBFGS, NAdam, RAdam and Adadelta, remove DecayedAdagrad, FtrlOptimizer and ModelAverage.

- https://github.com/PaddlePaddle/Paddle/issues/71203

@DrRyanHuang 和 @sunzhongkai588

### CODE REVIEW

**Comment by cangtianhuang** [comment]
文档预览链接为：http://preview-pr-7106.paddle-docs-preview.paddlepaddle.org.cn/documentation/docs/en/develop/api_guides/low_level/optimizer_en.html

File: docs/api_guides/low_level/optimizer_en.rst [hunk 1]
```diff
@@ -8,102 +8,106 @@ Neural network in essence is a `Optimization problem <https://en.wikipedia.org/w
...
 
-API Reference: :ref:`api_fluid_optimizer_SGDOptimizer`
+API Reference: :ref:`api_paddle_optimizer_Adam`
 
 
-2.Momentum/MomentumOptimizer
-----------------------------
+2. SGD
+------------------------
+:code:`SGD` is an offspring class of :code:`Optimizer` implementing `Random Gradient Descent <https://arxiv.org/pdf/1609.04747>`_ which is a method of `Gradient Descent <https://en.wikipedia.org/wiki/Gradient_descent>`_ .
```
Comment (DrRyanHuang) [inline docs/api_guides/low_level/optimizer_en.rst]: `Random Gradient Descent` -> `Stochastic Gradient Descent`

File: docs/api_guides/low_level/optimizer_en.rst [hunk 2]
```diff
@@ -8,102 +8,106 @@ Neural network in essence is a `Optimization problem <https://en.wikipedia.org/w
...
 
-1.SGD/SGDOptimizer
-------------------
 
-:code:`SGD` is an offspring class of :code:`Optimizer` implementing `Random Gradient Descent <https://arxiv.org/pdf/1609.04747.pdf>`_ which is a method of `Gradient Descent <https://zh.wikipedia.org/zh-hans/gradient_descent_algorithm>`_ .
-When it needs to train a large number of samples, we usually choose :code:`SGD` to make loss function converge more quickly.
+1. Adam
+------------------------
+Optimizer of `Adam <https://arxiv.org/abs/1412.6980>`_ is a method to adaptively adjust learning rate,
+fit for most  non- `convex optimization <https://en.wikipedia.org/wiki/Convex_optimization>`_ , big data set and high-dimensional scenarios. :code:`Adam` is the most common optimization algorithm.
```
Comment (DrRyanHuang) [inline docs/api_guides/low_level/optimizer_en.rst]:
```
non-`convex optimization` 修改为 `non-convex optimization`
```

非`凸优化` 修改为 `非凸优化`，麻烦同学中文部分也修改一下~

**Review by DrRyanHuang — CHANGES_REQUESTED** [review]

File: docs/api_guides/low_level/optimizer_en.rst (hunk 1 above)
Comment (cangtianhuang) [reply docs/api_guides/low_level/optimizer_en.rst]: indeed！

File: docs/api_guides/low_level/optimizer_en.rst (hunk 2 above)
Comment (cangtianhuang) [reply docs/api_guides/low_level/optimizer_en.rst]: 这里的 `convex optimization` 应该对应的是 wiki 链接： https://en.wikipedia.org/wiki/Convex_optimization ，且 wiki 中没有非凸优化的词条。我认为这样是合理的😂

Comment (DrRyanHuang) [reply docs/api_guides/low_level/optimizer_en.rst]: 这里主要纠结断句方式，是`非` `凸优化` 还是 `非凸优化`，神经网络的损失函数是一个 `非凸` 的，解决 `非凸` 函数的优化问题，被称为 `非凸优化`

关于 `非凸优化` 的其他内容，可以参考 [cs6787 的 PPT](https://www.cs.cornell.edu/courses/cs6787/2017fa/Lecture7.pdf)

Comment (cangtianhuang) [reply docs/api_guides/low_level/optimizer_en.rst]: 我明白您的意思了。我将此链接改为该ppt的链接可否？

**Review by DrRyanHuang — APPROVED** [review]
LGTM
预览[链接🔗](http://preview-pr-7106.paddle-docs-preview.paddlepaddle.org.cn/documentation/docs/en/develop/api_guides/low_level/optimizer_en.html)
